.github/workflows/on-pull-request-test.yaml

```diff
@@ -0,0 +1,37 @@
+
+name: Test
+on:
+  pull_request:
+
+jobs:
+  tests:
+    name: Run linter and tests
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 16
+
+      - name: Install dependencies
+        run: yarn install
```

Comment:
`yarn install` will be run twice for each PR update.  It is almost worth putting the test as a prerequisite on the build and then using caching on node_modules to avoid the second download.